### PR TITLE
Added book entity simple logic

### PR DESCRIPTION
**Pull request by pikador10**

### CODE REVIEW

File: src/main/java/com/hw/bookstore/domain/entity/Book.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Entity
+@Table(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", updatable = false, nullable = false)
+    private Long id;
+
+    @NotNull
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/domain/entity/Book.java]: Use the annotation Column(nullable = false). NotNull you should use with requestDto object  (the same for other fields)

File: src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import lombok.RequiredArgsConstructor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java:15]: Let's use JPA repository implementation ;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution ;)

File: src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java (hunk 2 above)
Comment (liliia-ponomarenko) [reply src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java:15]: I see that you have switched to using JPARepository in the third HW, so you can ignore this comment

File: src/main/java/com/hw/bookstore/domain/entity/Book.java (hunk 1 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/domain/entity/Book.java]: I am happy to see your comments again)
Fixed

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Pull request merged**